airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -189,17 +225,39 @@ def test_read_sequential_slices(
         # We sometimes have duplicate identical state messages in a stream which we can filter out to speed things up
         checkpoint_messages = [message for index, message in enumerate(checkpoint_messages) if message not in checkpoint_messages[:index]]
 
-        # To avoid spamming APIs we only test a fraction of slices
-        num_slices_to_test = 1 if len(checkpoint_messages) <= 5 else len(checkpoint_messages) // 5
-        for message_batch in checkpoint_messages[::num_slices_to_test]:
+        # To avoid spamming APIs we only test a fraction of batches (10%) and enforce a minimum of 10 tested
+        min_batches_to_test = 10
+        sample_rate = len(checkpoint_messages) // min_batches_to_test
+        stream_name_to_per_stream_state = dict()
+        for idx, message_batch in enumerate(checkpoint_messages):
             assert len(message_batch) > 0 and message_batch[0].type == Type.STATE
-            current_state = message_batch[0]
-            output = docker_runner.call_read_with_state(connector_config, configured_catalog_for_incremental, current_state.state.data)
+
+            if is_per_stream:
```

Comment:
the body of this test is getting difficult to read. Could we hide the conditional complexity in a helper method?

Reply:
I moved this logic to a specific helper function.